daml-script/export/src/test/scala/com/daml/script/export/EncodeValueSpec.scala

```diff
@@ -88,9 +94,25 @@ class EncodeValueSpec extends AnyFreeSpec with Matchers {
       encodeValue(Map.empty, Map(ContractId("my-contract-id") -> "mapped_cid"), cid.sum)
         .render(80) shouldBe "mapped_cid"
     }
-    "list" in {
-      val l = v.Value().withList(v.List(Seq(v.Value().withInt64(0), v.Value().withInt64(1))))
-      encodeValue(Map.empty, Map.empty, l.sum).render(80) shouldBe "[0, 1]"
+    "simple list" in {
+      val l = List(0L, 1L).map(v.Value().withInt64(_))
+      val value = v.Value().withList(v.List(l))
+      encodeValue(Map.empty, Map.empty, value.sum).render(80) shouldBe "[0, 1]"
+    }
+    "list of record" in {
+      val l = List(record, record).map(v.Value().withRecord(_))
+      val value = v.Value().withList(v.List(l))
+      encodeValue(Map.empty, Map.empty, value.sum).render(
+        100
+      ) shouldBe "[(M.R1 {a = 1, b = (M.R2 {c = 42}), c = M.R3}), (M.R1 {a = 1, b = (M.R2 {c = 42}), c = M.R3})]"
```

Comment:
without my change the output is 
```
[M.R1 with
  a = 1
  b = M.R2 with
    c = 42
  c = M.R3, M.R1 with
  a = 1
  b = M.R2 with
    c = 42
  c = M.R3]
```
which is not syntactically correct.

Reply:
I followed @sofiafaro-da advice to use curly braces notation which is easier to ident.

Reply:
We actually considered using curly braces initially but decided against it because the output of an export is intended to be read and for better or for worse, we push users towards `with`-syntax so using someting different in exports seemed off. I don’t feel too strongly about it but I think it’s at least worth thinking about whether we cannot fix the indentations instead.

Reply:
Now I'm thinking that the problem is actually in the parser, I don't see why the current output 

```haskell
[M.R1 with
  a = 1
  b = M.R2 with
    c = 42
  c = M.R3, M.R1 with
  a = 1
  b = M.R2 with
    c = 42
  c = M.R3]
```

shouldn't be syntactically correct. It reminds me of `do` notation, where the actions can be less indented than the `do` keyword, plus `BlockArguments`, where a function can appear just before the `do` (no `$` required). Which is to say, maybe we should fix the parser to support this syntax instead?

Reply:
@cocreature I disagree. We don't have a daml formatter (or even an AST) so trying to create particularly "human friendly" output is not a realistic goal. Moving to curly braces syntax here is fine, especially if it fixes a bug and is easier to maintain.

Reply:
I’m not that concerned about particularly nice formatting but using syntax that we don’t really use anywhere else in our docs just seems to make it unnecessarily hard for users to understand the output.
I don’t feel very strongly here though so feel free to move forward with this change.